### DISCUSSION TITLE
Using Tile Layer for non square Tiles (non regular Data)

### DISCUSSION THREAD

**Discussion by JanVanDerWall** [visgl/deck.gl#6558] [Q&A]
Disclaimer:
I am sorry if the question is obvious for some people. I am new to deck.gl and GeoData visualization.

I am trying to use the Tile Layer in Combination with a Bitmap Layer for the vitalization of high resolution raster Data. The Problem is, the Data is present in a lon/lat grid. I've configured my backend in a way, that is returns the right rectangles of data. But, due to the nature of the Data the width of the rectangle is fixed (540) while the height is changing depending on the tile selected. Which means the resolution of the tiles is not the same for every tile. As far as I understand the tile layer expects tiles with the same width and height since the OSM tiles are that way. The discrepancy leads to a wrong display of the data. Is it possible to use the Tile Layer in that way, or do I need to look for a different solution.

Thanks!

My Code:
```
new TileLayer({
    id: "data",
    data: "http://127.0.0.1:8000/tiled_data/?x={x}&y={y}&z={z}",
    minZoom: 0,
    maxZoom: 4,
    tileSize: 540,
    renderSubLayers: props => {
        const {
            bbox: { west, south, east, north }
        } = props.tile;

        return new BitmapLayer(props, {
            data: null,
            image: {
                width: props.data[1][1],
                height: props.data[1][0],
                format: GL.RGBA,
                data: generateImgData(props.data)
            },
            _imageCoordinateSystem: COORDINATE_SYSTEM.LNGLAT,
            bounds: [west, south, east, north]
        });
    }
}),
```

GenerateImgData creates an RGBA array from the data provided by the backend

![Screenshot from 2022-01-14 14-02-28](https://user-images.githubusercontent.com/56517555/149519601-4a498327-e0de-478f-a037-150205a789f0.png)

**Comment by Pessimistress** [comment]
https://github.com/visgl/deck.gl/pull/5504

**Reply by JanVanDerWall** [reply]
I don't quite understand yet. Is there a native way to use other projections or coordinate systems (WSG 84) yet?

**Reply by kylebarron** [reply]
My understanding is that deck.gl supports other coordinate systems for some layers, like the `BitmapLayer`, but right now the TileLayer only supports the Web Mercator indexing system. #5504 discusses how we might allow users to use other indexing systems without putting too much of a maintenance burden on deck.gl.

**Reply by JanVanDerWall** [reply]
Sorry I might have framed my question the wrong way. I was asking: Am I right with my assumption that this is not available and tested yet. If I am wrong with that assumption I did not find any documentation concerning it. Also If I am right, Is there no way another tile system can be achieved without a lot of work right now?

**Reply by kylebarron** [reply]
You're correct that this is not available right now. I think the easiest way to go about this in deck.gl right now would be to modify the internals of the TileLayer to fix where the tiles are.

**Reply by JanVanDerWall** [reply]
Thank you very much. I will try to do so, and share my results.
